lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
In long run this is not a great idea. I did some analysis with 500 GB of
data , 50 million dims and 400 mappers. It did not make any difference in
cost per iteration. But then time per iteration was 30 seconds and hence gc
time might be neglegible compared to that. But then with big data these are
pretty much the times we are looking at.
On Sep 27, 2015 7:07 PM, "Markus Weimer" <notifications@github.com> wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs
> <https://github.com/apache/incubator-reef/pull/527#discussion_r40511105>:
>
> >              {
> > +                GC.Collect();
>
> This looks dangerous, as it will incur some cost. @anupam128
> <https://github.com/anupam128> @afchung <https://github.com/afchung> Can
> you comment on this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/527/files#r40511105>.
>